pennylane/optimize/qnspsa.py

```diff
@@ -0,0 +1,407 @@
+    ...     qml.RX(params[0], wires=0)
+    ...     qml.CRY(params[1], wires=[0, 1])
+    ...     return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+    Once constructed, the qnode can be passed directly to the ``step`` or ``step_and_cost``
+    function of the optimizer.
+
+    >>> params = np.random.rand(2)
+    >>> opt = QNSPSAOptimizer(stepsize=5e-2)
+    >>> for i in range(51):
+    >>> params, loss = opt.step_and_cost(cost, params)
+    >>> if i % 10 == 0:
+    ...     print(f"Step {i}: cost = {loss:.4f}")
+    Step 0: cost = 0.9987
+    Step 10: cost = 0.9841
+    Step 20: cost = 0.8921
+    Step 30: cost = 0.0910
+    Step 40: cost = -0.9369
+    Step 50: cost = -0.9984
+
+    Keyword Args:
+        stepsize (float): the user-defined hyperparameter :math:`\eta` for learning rate
+        regularization (float): regularitzation term :math:`\beta` to the Fubini-Study metric tensor
+            for numerical stability
+        finite_diff_step (float): step size :math:`\epsilon` to compute the finite difference
+            gradient and the Fubini-Study metric tensor
+        resamplings (int): the number of samples to average for each parameter update
+        blocking (boolean): when set to be True, the optimizer only accepts updates that lead to a
+            loss value no larger than the loss value before update, plus a tolerance. The tolerance
+            is set with the hyperparameter ``history_length``. The ``blocking`` option is
+            observed to help the optimizer to significantly converge faster
+        history_length (int): when ``blocking`` is True, the tolerance is set to be the average of
+            the cost values in the last ``history_length`` steps
+        seed (int): seed for the random sampling
+    """
+    # pylint: disable=too-many-arguments
+    # pylint: disable=too-many-instance-attributes
+    def __init__(
+        self,
+        stepsize=1e-3,
+        regularization=1e-3,
+        finite_diff_step=1e-2,
+        resamplings=1,
+        blocking=True,
+        history_length=5,
+        seed=None,
+    ):
+        self.stepsize = stepsize
+        self.reg = regularization
+        self.finite_diff_step = finite_diff_step
+        self.metric_tensor = None
+        self.k = 1
+        self.resamplings = resamplings
+        self.blocking = blocking
+        self.last_n_steps = np.zeros(history_length)
+        self.rng = np.random.default_rng(seed)
+
+    def step(self, cost, *args, **kwargs):
+        """Update trainable arguments with one step of the optimizer.
+
+        .. note::
+            When blocking is set to be True, ``step`` calls ``step_and_cost`` on the backend, as loss
+            measurements are required by the algorithm in this scenario.
+
+        Args:
+            cost (QNode): the QNode wrapper for the objective function for optimization
+            *args : variable length argument list for qnode
+            **kwargs : variable length of keyword arguments for the qnode
```

Comment:
It's standard to exclude these asterisks in the arguments list.
```suggestion
            args : variable length argument list for qnode
            kwargs : variable length of keyword arguments for the qnode
```